bot/exts/evergreen/trivia_quiz.py

```diff
@@ -264,7 +348,7 @@ async def quiz_game(self, ctx: commands.Context, category: Optional[str], questi
         topic_length = len(topic)
 
         if questions is None:
-            self.question_limit = DEFAULT_QUESTION_LIMIT
+            self.question_limit = min(DEFAULT_QUESTION_LIMIT, topic_length)
```

Comment:
This still doesn't protect against the infinite loop, seemingly because of an off-by-one bug throughout this cog. A simple fix for this would probably be just to use `min(DEFAULT_QUESTION_LIMIT, topic_length-1)`, or you could look into this further and try and fix this off-by-one in the rest of the cog (I think the root cause is line 389 `if len(done_questions) > self.question_limit and hint_no == 0:` which should instead be `>=`, as if the number of completed questions is greater than the limit, we've already done too many).